src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -383,24 +383,28 @@ def _await_fluent_launch(
 
 
 def _get_server_info(
-    server_info_file_name: str,
+    server_info_file_name: Optional[str] = None,
     ip: Optional[str] = None,
     port: Optional[int] = None,
     password: Optional[str] = None,
 ):
     """Get server connection information of an already running session."""
-    if ip and port:
+    if not (ip and port) and not server_info_file_name:
+        raise RuntimeError(
+            "Please provide either ip and port data or server-info file."
+        )
+    if ip and port and server_info_file_name:
```

Comment:
This could be

`if (ip or port) and server_info_file_name:`

Reply:
:100:

Reply:
@mkundu1 https://github.com/ansys/pyfluent/pull/2176/files#diff-e49ebe5caff347c437627b296d498b5c53873625992b95779b6bacb1965e5f07.